src/test/HostActivationTests/NativeHosting/ComponentActivation.cs

```diff
@@ -0,0 +1,92 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.DotNet.Cli.Build.Framework;
+using System.IO;
+using Xunit;
+
+namespace Microsoft.DotNet.CoreSetup.Test.HostActivation.NativeHosting
+{
+    public class Component
+    {
+        public static int ComponentEntryPoint(System.IntPtr arg, int size)
+        {
+            System.Console.WriteLine($"Called ComponentEntryPoint(0x{arg.ToString("x")}, {size})");
+
+            return size >> 1;
+        }
+    }
+
+    public partial class ComponentActivation : IClassFixture<ComponentActivation.SharedTestState>
+    {
+        private const string CreateNativeDelegateArg = "component_create_native_delegate";
+        
+        private readonly SharedTestState sharedState;
+
+        public ComponentActivation(SharedTestState sharedTestState)
+        {
+            sharedState = sharedTestState;
+        }
+
+        //[Theory]
+        //[InlineData(true,  true,  true)]
+        //[InlineData(false, true,  true)]
+        //[InlineData(true,  false, true)]
+        //[InlineData(true,  true,  false)]
+        public void CallDelegate(bool validPath, bool validType, bool validMethod)
+        {
+            string[] args =
+            {
+                CreateNativeDelegateArg,
+                sharedState.HostFxrPath,
+                sharedState.RuntimeConfigPath,
+                validPath ? typeof(Component).Assembly.Location : "BadPath...",
```

Comment:
This feels wrong - we're loading the XUnit test assembly here. I guess it will work, but lot of its dependencies will probably not resolve (there's no `.deps.json` for it). It's not testing anything like the main scenario. Why can't we use one of the component assemblies in the test projects?

Reply:
Ergh, I had switched it to target a separate component. And then decided to try it in the test assembly to see what issue Steve might have been hitting. Apparently I also committed that in the change to comment out the tests...